src/v/v8_engine/script.cc

```diff
@@ -92,9 +137,31 @@ void script::set_function(std::string_view name) {
     _function.Reset(_isolate.get(), function_val.As<v8::Function>());
 }
 
-// It is the first version for run. In next updates I use alient_thread for run
-// v8.
-void script::run(ss::temporary_buffer<char>& data) { run_internal(data); }
+ss::future<> script::run(ss::temporary_buffer<char>& data) {
+    return ss::with_semaphore(_mutex, 1, [this, &data] {
+        return _executor
+          .submit(
+            [this] {
+                _is_cancel = false;
+                _watchdog.rearm(seastar::lowres_clock::time_point(
+                  seastar::lowres_clock::now() + _timeout_ms));
+            },
+            [this, &data] { run_internal(data); })
+          .handle_exception_type([this](const ss::sleep_aborted&) {
+              throw_exception_from_v8("Executor is closed");
+          })
+          .handle_exception_type([this](const ss::gate_closed_exception&) {
+              throw_exception_from_v8("Executor is closed");
+          })
+          .finally([this] {
+              if (!_is_cancel) {
+                  _watchdog.cancel();
+              } else {
+                  continue_execution();
```

Comment:
why must we call continue every time?

Reply:
If we stopped v8::script - yes

Reply:
Maybe the name is not so good. It cancel terminate execution for isolate. We want to reuse this isolate in future for next executions

Reply:
done